types/aws-lambda/trigger/codecommit.d.ts

```diff
@@ -0,0 +1,36 @@
+import { Handler } from "../handler";
+
+export type CodeCommitHandler = Handler<CodeCommitTrigger, void>;
+
+/**
+ * CodeCommit events
+ * https://docs.aws.amazon.com/lambda/latest/dg/services-codecommit.html
+ */
+export interface CodeCommitTrigger {
+    awsRegion: string;
+    codecommit: {
+        references: Array<
+        {
+            commit: string;
+            created?: boolean;
+            deleted?: boolean;
+            ref: string;
+        }>;
+    };
+    customData: string;
```

Comment:
I'm wondering whether this one should be optional given it's not in the example event in the AWS docs, unless it's set to an empty string constantly if a value isn't provided whilst making the event? https://docs.aws.amazon.com/codecommit/latest/userguide/how-to-notify-lambda.html

```suggestion
    customData?: string;
```

(AWS documentation can be quite sparse on specifics so I might be wrong!)

Reply:
In the tests I made the incoming event always has `customData: ""` even though I configured the trigger with empty custom data field, so I assume it's always present. I couldn't find any specification on that event format, alas....

Reply:
I can still change it to optional if it suits you better

Reply:
I think optional might be better, cheers! I had a quick play and got customData as a JSON null in my account:
```
{
  "Records": [
    {
      "awsRegion": "eu-west-2",
      "codecommit": {
        "references": [
          {
            "commit": "25f945a6135b70ffb93b3476ed1615f157b7c011",
            "ref": "refs/heads/main"
          }
        ]
      },
      "customData": null,
      "eventId": "72f89f41-d36c-49e0-9719-e068b2284556",
      "eventName": "ReferenceChanges",
      "eventPartNumber": 1,
      "eventSource": "aws:codecommit",
      "eventSourceARN": "arn:aws:codecommit:eu-west-2:869036666965:test-repo",
      "eventTime": "2023-03-13T18:15:49.572+0000",
      "eventTotalParts": 1,
      "eventTriggerConfigId": "4037999f-048a-4ff3-a479-d7f0fd8b6671",
      "eventTriggerName": "TestTrigger",
      "eventVersion": "1.0",
      "userIdentityARN": "arn:aws:sts::869036666965:assumed-role/AWSReservedSSO_AWSAdministratorAccess_blah"
    }
  ]
}
```